rclpy/test/test_qos.py

```diff
@@ -124,3 +126,65 @@ def test_preset_profiles(self):
         assert (
             QoSPresetProfiles.SYSTEM_DEFAULT.value ==
             QoSPresetProfiles.get_from_short_key('system_default'))
+
+
+class TestCheckQosCompatibility(unittest.TestCase):
+
+    def test_compatible(self):
+        for durability in [QoSDurabilityPolicy.VOLATILE, QoSReliabilityPolicy.RELIABLE]:
+            qos = QoSProfile(
+                depth=1,
+                durability=durability,
+                lifespan=Duration(seconds=1),
+                deadline=Duration(seconds=1),
+                liveliness=QoSLivelinessPolicy.AUTOMATIC,
+                liveliness_lease_duration=Duration(seconds=1),
+            )
```

Comment:
@jacobperron, is this what you meant in https://github.com/ros2/rclpy/pull/708#discussion_r597353191?

Reply:
Sorry, I was suggesting we add a `reliability` keyword arg like this:
```diff
qos = QoSProfile(
                depth=1,
+               reliability=QoSReliabilityPolicy.RELIABLE,
+               durability=QoSDurabilityPolicy.VOLATILE,
-               durability=durability,
                lifespan=Duration(seconds=1),
                deadline=Duration(seconds=1),
                liveliness=QoSLivelinessPolicy.AUTOMATIC,
                liveliness_lease_duration=Duration(seconds=1),
            )
```

Note *durability* and *reliability* are difference policy kinds. The current code as written is technically wrong (assigning `QoSReliabilityPolicy.RELIABLE` as the durability).

Reply:
> Note *durability* and *reliability* are different policy kinds. The current code as written is technically wrong (assigning `QoSReliabilityPolicy.RELIABLE` as the durability).

Ah, I get it. I was thinking of them as different types of policies. Thanks for the correction.